File: app/models/butterfly.rb
class Butterfly 
  include Mongoid::Document
  field :name, type: String
  field :description, type: String
  field :pointValue, type: Integer
  field :image, type: String
  field :caught, type: Boolean, default: false
  belongs_to :game

  store_in collection: "butterflies", database: "app24394627"

end

File: app/assets/javascripts/backbone/models/butterfly.js
var Butterfly = Backbone.Model.extend({
  idAttribute: "_id",

  defaults: {
    name: null,
    description: null,
    pointValue: null,
    caught: false,
    image: null,
    game_id: undefined
  },

  catch: function() {
    this.set({ caught: true });
    this.save();
  },

  urlRoot: function() {
    return '/games/'+ gameID + '/butterflies';
  }
})

